src/session.rs

```diff
@@ -58,30 +57,14 @@ impl Session<SecretConnection<TcpStream>> {
 }
 
 impl Session<UnixConnection<UnixStream>> {
-    pub fn accept_unix(chain_id: chain::Id, socket_path: &Path) -> Result<Self, KmsError> {
-        // Try to unlink the socket path, shouldn't fail if it doesn't exist
-        if let Err(e) = fs::remove_file(socket_path) {
-            if e.kind() != io::ErrorKind::NotFound {
-                return Err(KmsError::from(e));
-            }
-        }
-
-        debug!(
-            "{}: Waiting for a connection at {}...",
-            chain_id,
-            socket_path.to_str().unwrap()
-        );
-
-        let listener = UnixListener::bind(&socket_path)?;
-        let (socket, addr) = listener.accept()?;
-
-        debug!("{}: Accepted connection from {:?}", chain_id, addr);
+    pub fn connect_unix(chain_id: chain::Id, socket_path: &Path) -> Result<Self, KmsError> {
         debug!(
-            "{}: Stopped listening on {}",
+            "{}: Connecting to socket at {}...",
             chain_id,
             socket_path.to_str().unwrap()
         );
 
+        let socket = UnixStream::connect(socket_path).unwrap();
```

Comment:
Oh, sorry, I missed that one: The `unwrap` here should be replaced by sth that returns the Err in an error case. Can be replaced by `UnixStream::connect(socket_path)?` (the error will be converted to error::KmsError automagically).

Reply:
Yes, you're right - let me fix that quick 👍

Reply:
Ideally I would suggest making code as panic-free as possible. IMO panics should only occur in buggy code where invariants have been violated, and not for things like I/O errors.

Reply:
Thanks Tony! This was updated (and merged) immediately in #150.

Reply:
Great!